webpack/assets/javascripts/react_app/routes/Models/ModelsPage/ModelsPage.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'patternfly-react';
+import { Link } from 'react-router-dom';
+
+import { translate as __ } from '../../../common/I18n';
+import PageLayout from '../../common/PageLayout/PageLayout';
+import ModelsPageContent from './components/ModelsPageContent';
+import { MODELS_SEARCH_PROPS } from '../constants';
+
+const ModelsPage = ({
+  fetchAndPush,
+  search,
+  isLoading,
+  hasData,
+  models,
+  page,
+  perPage,
+  sort,
+  hasError,
+  itemCount,
+  message,
+  canCreate,
+}) => {
+  const handleSearch = query => fetchAndPush({ searchQuery: query, page: 1 });
+
+  const createBtn = (
+    <Link to="/models/new">
+      <Button bsStyle="primary">{__('Create Model')}</Button>
+    </Link>
+  );
+
+  return (
+    <PageLayout
+      header={__('Hardware Models')}
+      searchable
+      searchProps={MODELS_SEARCH_PROPS}
+      searchQuery={search}
+      isLoading={isLoading && hasData}
```

Comment:
maybe not in the scope of this PR, but a 100ms delay before showing the little spinner could be very nice